packages/plugin-betswirl/src/utils/betswirl.ts

```diff
@@ -0,0 +1,307 @@
+        const chainlinkVRFCostContract = new ethers.Contract(
+            chainlinkVRFCostFunctionData.data.to,
+            chainlinkVRFCostFunctionData.data.abi,
+            provider
+        );
+        const chainlinkVRFCost: bigint = await chainlinkVRFCostContract[
+            chainlinkVRFCostFunctionData.data.functionName
+        ](...chainlinkVRFCostFunctionData.data.args, {
+            gasPrice,
+        });
+        return chainlinkVRFCost;
+    } catch (error) {
+        throw new Error(
+            `An error occured while getting the chainlink vrf cost: ${error.shortMessage}`
+        );
+    }
+}
+
+export async function placeBet(
+    chainId: CasinoChainId,
+    walletClient: MoxieWalletClient,
+    game: CASINO_GAME_TYPE,
+    gameEncodedInput: GameEncodedInput,
+    gameMultiplier: number,
+    casinoGameParams: {
+        betAmount: bigint;
+        betToken: Token;
+        betCount: number;
+        receiver: Hex;
+        stopGain: bigint;
+        stopLoss: bigint;
+    }
+) {
+    const betRequirements = await getBetRequirements(
+        chainId,
+        game,
+        casinoGameParams.betToken,
+        gameMultiplier
+    );
+
+    if (!betRequirements.isAllowed) {
+        throw new Error(`The token isn't allowed for betting`);
+    }
+    if (casinoGameParams.betAmount > betRequirements.maxBetAmount) {
+        throw new Error(
+            `Bet amount should be less than ${betRequirements.maxBetAmount}`
+        );
+    }
+    if (casinoGameParams.betCount > betRequirements.maxBetCount) {
+        throw new Error(
+            `Bet count should be less than ${betRequirements.maxBetCount}`
+        );
+    }
+
+    const functionData = getPlaceBetFunctionData(
+        {
+            betAmount: casinoGameParams.betAmount,
+
+            game,
+            gameEncodedInput: gameEncodedInput,
+            receiver: casinoGameParams.receiver,
+            betCount: casinoGameParams.betCount,
+            tokenAddress: casinoGameParams.betToken.address,
+            stopGain: casinoGameParams.stopGain,
+            stopLoss: casinoGameParams.stopLoss,
+        },
+        chainId
+    );
+
+    try {
+        await approveIfERC20(
+            walletClient,
+            chainId,
+            game,
+            casinoGameParams.betToken,
+            casinoGameParams.betAmount
+        );
+
+        const provider = new ethers.JsonRpcProvider("https://mainnet.base.org");
+        const gasPrice = ((await provider.getFeeData()).gasPrice * 120n) / 100n;
+        const vrfCost =
+            ((await getChainlinkVrfCost(
+                chainId,
+                game,
+                casinoGameParams.betToken.address,
+                casinoGameParams.betCount,
+                gasPrice
+            )) *
+                120n) /
+            100n;
+        const { hash: betHash } = await walletClient.sendTransaction(
+            chainId.toString(),
+            {
+                toAddress: functionData.data.to,
```

Comment:
Add this to make it work on prod, as looks like there's additional checks to include this:
```suggestion
                fromAddress: walletClient.address,
                toAddress: functionData.data.to,
```